### PR TITLE
Improve implementation to support encoding and Drupal's SHA-512 '$S$' mode

### PR DESCRIPTION
**Pull request by symbioquine**
Thanks for creating this password hash provider @fjf2002!

To work for my use-case I needed to be able to import Drupal's hashes which use a modified PHPass `$S$` SHA-512 mode. It was also useful for testing to have the ability to encode passwords so I added that too. (Though it is still up to the wise sysadmin to migrate away from the PHPass hashing to the superior `pbkdf2-sha256` in good time.)

Support for Drupal's SHA-512 password hashing is roughly based on the description at http://joncave.co.uk/2011/01/password-storage-in-drupal-and-wordpress/ and details from https://github.com/drupal/drupal/blob/3515f040ae71496c4e4387e44fa2afd963762252/core/lib/Drupal/Core/Password/PhpassHashedPasswordBase.php

### CODE REVIEW

**Comment by fjf2002** [comment]
Hello! Thank You for participating.

I had created this password hash provider to support PHPass hashes, just like

- https://github.com/leroyguillaume/keycloak-bcrypt exists to handle bcrypt hashes,
- or https://github.com/fjf2002/keycloak-md5 handles md5 hashes.

**I consider PHPass and MD5 hashed as outdated.** So I decided to support just password verification, not encryption/hashing of new passwords. Especially I do not want to enable or encourage anybody to encrypt new passwords using these old algorithms.

In our Keycloak production system, we use this to be able to migrate those old hashes from other systems to Keycloak. However at first login, Keycloak will re-encrypt those hashes using pbkdf-sha256. (This is archieved setting Authentication->Policies->Hashing Algorithm to pbkdf2-sha256.) Finally we will deprecate those old hashing algorithms by removing the corresponding providers from Keycloak by the end of this year.

Long story short,
- I consider offering encoding capabilities for PHpass hashes at least careless, if not a security problem.
- You should therefore use you own repo, I think?

Thank You.
